library/channel.c

```diff
@@ -918,8 +914,23 @@ static void on_channel_data(uv_stream_t *s, ssize_t len, const uv_buf_t *buf) {
 
 static void on_channel_connect_internal(uv_connect_t *req, int status) {
     tlsuv_stream_t *tls = (tlsuv_stream_t *)req->handle;
+
+    // connect request was cancelled via tlsuv_stream_close
+    // cleanup in close callback
+    if (status == UV_ECANCELED || tls->data == NULL) {
+        goto done;
+    }
+
     ziti_channel_t *ch = tls->data;
 
+    if (tls != ch->connection) {
```

Comment:
should `ch` still be checked for null?

Reply:
not really necessary, we always set it to `ch`